### DISCUSSION TITLE
Using system libraries

### DISCUSSION THREAD

**Discussion by phpaiva** [pyinstaller/pyinstaller#8838] [Help]
I can't find a way to compile just the interpreter and the project, without the libraries in the system.

I mean, to use the /usr/lib directory as usual, not packing into a standalone or by a dedicated folder with clones of the libraries. I want to take advantage of the shared libraries in the same way the application works without being compiled with pyinstaller.

Thanks in advance.

**Comment by rokm** [comment]
There is `Analysis.exclude_system_libraries()` that might do the trick: https://pyinstaller.org/en/stable/spec-files.html#posix-specific-options

But on the other hand, if you aren't looking to create a stand-alone bundle, why use PyInstaller in the first place?

**Reply by phpaiva** [reply]
> But on the other hand, if you aren't looking to create a stand-alone bundle, why use PyInstaller in the first place?

Just to cover the source code a little. I work on OpenWRT and MIPS architecture. Getting an obfuscator or compiler up and running has been an impossible challenge with the exception of pyinstaller. I wanted to make a text obfuscator since there are no useful obfuscators for complex projects, but I realized that the work was getting really complicated and I don't have time to overthink this either.

So I settle for covering the code a little to prevent hands with very little knowledge from trying to degenerate the spirit of my initiative, which at the moment I am not interested in spreading too much.

I tried to test what you proposed but it still going for the "_internal" folder:

[PYI-2597:ERROR] Failed to load Python shared library '/opt/_internal/libpython3.11.so.1.0': 

Here is the .spec file:
```
# -*- mode: python ; coding: utf-8 -*-


a = Analysis(
    ['mesh.py'],
    pathex=[],
    binaries=[],
    datas=[],
    hiddenimports=[],
    hookspath=[],
    hooksconfig={},
    runtime_hooks=[],
    excludes=[],
    noarchive=False,
    optimize=0,
)
pyz = PYZ(a.pure)

a.exclude_system_libraries()

exe = EXE(
    pyz,
    a.scripts,
    [],
    exclude_binaries=True,
    name='mesh',
    debug=False,
    bootloader_ignore_signals=False,
    strip=False,
    upx=True,
    console=True,
    disable_windowed_traceback=False,
    argv_emulation=False,
    target_arch=None,
    codesign_identity=None,
    entitlements_file=None,
)
coll = COLLECT(
    exe,
    a.binaries,
    a.datas,
    strip=False,
    upx=True,
    upx_exclude=[],
    name='mesh',
)

```

**Reply by rokm** [reply]
> [PYI-2597:ERROR] Failed to load Python shared library '/opt/_internal/libpython3.11.so.1.0':

Even with `Analysis.exclude_system_libraries()`, you cannot *completely* get rid of `_internal` directory. And python shared library should still be bundled (and the stdlib binary extensions as well)... If you want to get rid of everything, you will need to look into customizing bootloader code.

**Reply by bwoodsend** [reply]
> just to cover the source code a little.

You might as well just `python -m py_compile` then remove your source code and leave it at that then. That's the only function of PyInstaller you'd be using.

**Reply by Harshith9898** [reply]
hi i am also trying to obfuscation can u pls help me on this

**Reply by bwoodsend** [reply]
No, obfuscation is not something we want anything to do with.
